### PR TITLE
Feature/security

### PR DESCRIPTION
**Pull request by iimpulse**
Hi All,

Pretty basic implementation of Auth0 for angular + spring. Additionally we have saving this user into datastore. If you use spring active profile local it will use the datastore emulator and not connect to the cloud service. By default the cloud run service can connect to other services within the gcp project. Please review!

#207 #208 #199

### CODE REVIEW

File: phenopacketlab-restapi/src/main/java/org/monarchinitiative/phenopacketlab/restapi/domain/User.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	private String authId;
+
+	public User(){}
+
+	public User(String authId){
+		Assert.notNull(authId, "Required field authId not found.");
+		this.authId = authId;
+	}
+
+	public User(Authentication authentication){
```
Comment (ielis) [inline phenopacketlab-restapi/src/main/java/org/monarchinitiative/phenopacketlab/restapi/domain/User.java:24]: I think this constructor should be removed and the logic should be moved into `UserService`.

**Review by ielis — APPROVED** [review]
Hi @iimpulse thanks for the PR!

I think the code is OK and I'd like to suggest a few minor tweaks during our call today.
I haven't, however, looked into the frontend as it's not my cup of tea.

File: phenopacketlab-frontend/src/environments/environment.dev.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const URL = 'https://phenopacketlab-dev.jax.org';
-const API_URL = `${URL}/api/v1`;
+const API_URL = `${URL}/api`;
```
Comment (belkassaby) [inline phenopacketlab-frontend/src/environments/environment.dev.ts:2]: Let's remember to change all the routing in the load balancer for the backend redirection as api/v1/* was used there.

File: phenopacketlab-frontend/src/environments/environment.prod.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 const URL = 'https://phenopacketlab.jax.org';
-const API_URL = `${URL}/api/v1`;
+const API_URL = `${URL}/api`;
```
Comment (belkassaby) [inline phenopacketlab-frontend/src/environments/environment.prod.ts:2]: same comment as for the environment.dev: update load balancer after the deploy

File: phenopacketlab-restapi/src/main/java/org/monarchinitiative/phenopacketlab/restapi/Application.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package org.monarchinitiative.phenopacketlab.restapi;
 
+import com.google.cloud.spring.data.datastore.repository.config.EnableDatastoreRepositories;
```
Comment (belkassaby) [inline phenopacketlab-restapi/src/main/java/org/monarchinitiative/phenopacketlab/restapi/Application.java:3]: unused import

File: phenopacketlab-restapi/src/main/java/org/monarchinitiative/phenopacketlab/restapi/config/security/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
+package org.monarchinitiative.phenopacketlab.restapi.config.security;
+
+import com.google.cloud.spring.data.datastore.repository.config.EnableDatastoreRepositories;
```
Comment (belkassaby) [inline phenopacketlab-restapi/src/main/java/org/monarchinitiative/phenopacketlab/restapi/config/security/SecurityConfig.java:3]: unused import

File: phenopacketlab-restapi/src/main/resources/application-development.yml [hunk 6]
```diff
@@ -21,12 +21,20 @@ phenopacketlab:
     - 'HP'
 
 api:
-  version: /api/v1
+  version: /api
```
Comment (belkassaby) [inline phenopacketlab-restapi/src/main/resources/application-development.yml:24]: shouldn't we leave v1 here? as it relates to the version and not the path as used in the environment files?

File: phenopacketlab-frontend/src/app/app.component.ts [hunk 7]
```diff
@@ -15,26 +18,33 @@ import { SidebarComponent } from './component/sidebar/sidebar.component';
 })
 export class AppComponent implements OnInit {
```
Comment (belkassaby) [inline phenopacketlab-frontend/src/app/app.component.ts:19]: The app.component.spec.ts fails due to the AuthService provider missing I think. Let's fix it before merging into the development branch.

**Review by belkassaby — APPROVED** [review]

**Review by belkassaby — APPROVED** [review]

**Pull request merged**